### PR TITLE
feat: expose capability to get OCI image configuration.

### LINKED ISSUE #145: Update the Rust SDK adding a new host capability to get the container image config

### PR DESCRIPTION
**Pull request by jvanz**
## Description

Adds a new function to allow policy author to get the OCI image manifest and configuration from a given OCI image name.

Fix #145 

## Test
To test this pull request, you can run the following commands:

```shell
make test
```

### CODE REVIEW

File: src/host_capabilities/oci.rs [hunk 1]
```diff
@@ -24,6 +24,14 @@ pub enum OciManifestResponse {
     ImageIndex(Box<ImageIndex>),
 }
 
+/// Response to manifest digest request
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct OciManifestAndConfigtResponse {
```
Comment (viccuad) [inline src/host_capabilities/oci.rs:29]:
```suggestion
pub struct OciManifestAndConfigResponse {
```

File: src/host_capabilities/oci.rs [hunk 2]
```diff
@@ -44,8 +52,19 @@ pub fn get_manifest(image: &str) -> Result<OciManifestResponse> {
...
+    Ok(response)
+}
+
+/// Fetches OCI image manifest and configuration referenced by `image`
+pub fn get_manifest_and_config(image: &str) -> Result<OciManifestAndConfigtResponse> {
+    let req = json!(image);
+    let msg = serde_json::to_vec(&req)
+        .map_err(|e| anyhow!("error serializing the validation request: {}", e))?;
+    let response_raw = wapc_guest::host_call("kubewarden", "oci", "v1/oci_manifest_config", &msg)
+        .map_err(|e| anyhow!("error invoking wapc oci.manifest_digest: {:?}", e))?;
```
Comment (viccuad) [inline src/host_capabilities/oci.rs:65]:
```suggestion
        .map_err(|e| anyhow!("error invoking wapc oci.manifest_and_config: {:?}", e))?;
```

**Review by viccuad — APPROVED** [review]
Looks good, just some renames to do before merging.

File: src/host_capabilities/oci.rs [hunk 3]
```diff
@@ -44,8 +52,19 @@ pub fn get_manifest(image: &str) -> Result<OciManifestResponse> {
...
+
+/// Fetches OCI image manifest and configuration referenced by `image`
+pub fn get_manifest_and_config(image: &str) -> Result<OciManifestAndConfigtResponse> {
+    let req = json!(image);
+    let msg = serde_json::to_vec(&req)
+        .map_err(|e| anyhow!("error serializing the validation request: {}", e))?;
+    let response_raw = wapc_guest::host_call("kubewarden", "oci", "v1/oci_manifest_config", &msg)
+        .map_err(|e| anyhow!("error invoking wapc oci.manifest_and_config: {:?}", e))?;
+
+    let response: OciManifestAndConfigtResponse = serde_json::from_slice(&response_raw)?;
```
Comment (flavio) [inline src/host_capabilities/oci.rs:67]: there's a typo here

**Pull request merged**